File: java/src/main/java/command/receiver/XyzBank.java
package command.receiver;

import command.TransferParam;

import java.util.Objects;

public class XyzBank {
    public void transfer(TransferParam transferParam) {
        System.out.println("XyxBank 이체 시작");

        if (Objects.isNull(transferParam)) {
            throw new IllegalArgumentException("TransferParam required");
        }

        final var fee = calculateFee(transferParam.amount());
        final var transferAmount = transferParam.amount() - fee;

        var message = String.format("%.3f 원을 이체 했습니다.", transferAmount);
        System.out.println(message);
        System.out.println("XyxBank 이체 종료");
    }

    private double calculateFee(double amount) {
        return amount * 0.02;
    }
}

File: src/main/kotlin/command/receiver/XyzBank.kt
package command.receiver

import command.TransferParam

class XyzBank {
    fun transfer(transferParam: TransferParam) {
        println("XyzBank 이체 시작")

        val fee = calculateFee(transferParam.amount)
        val transferAmount = transferParam.amount - fee

        println("%.2f 원을 이체 했습니다.".format(transferAmount))
        println("XyzBank 이체 종료")
    }

    private fun calculateFee(amount: Double): Double {
        return amount * 0.02
    }
}